ToDoList/test/ToDoList.Api.Tests/Controllers/ToDosControllerTests.cs

```diff
@@ -0,0 +1,289 @@
+        {
+            // Arrange
+            _retrieveServiceSubstitute.RetrieveAllAsync().Returns(_toDoList);
+
+            // Act
+            var response = await _controller.ExecuteAction(controller => controller.GetToDosAsync());
+            response.TryGetContentValue(out IEnumerable<ToDo> result);
+
+            // Assert
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.OK), $"Expecting status code OK, but was {response.StatusCode}");
+            Assert.That(result, Is.EqualTo(_toDoList).UsingToDoComparer(), "Todos are not equal");
+            Assert.That(() => _retrieveServiceSubstitute.Received().RetrieveAllAsync(), Throws.Nothing, $"RetrieveAllAsync should have been called");
+        }
+
+        [Test]
+        public async Task GetToDoAsync_Exists_OkReturned()
+        {
+            // Arrange
+            const int itemIndex = 0;
+            var expectedToDo = _toDoList.ElementAt(itemIndex);
+            _retrieveServiceSubstitute.IsInDatabaseAsync(expectedToDo.Id).Returns(true);
+            _retrieveServiceSubstitute.RetrieveOneAsync(expectedToDo.Id).Returns(expectedToDo);
+
+            // Act
+            var response = await _controller.ExecuteAction(controller => controller.GetToDoAsync(expectedToDo.Id));
+            response.TryGetContentValue(out ToDo result);
+
+            // Assert
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.OK), $"Expecting status code OK, but was {response.StatusCode}");
+            Assert.That(result, Is.EqualTo(expectedToDo).UsingToDoComparer(), $"{result} is not equal to expected {expectedToDo}");
+            Assert.That(() => _retrieveServiceSubstitute.Received().RetrieveOneAsync(expectedToDo.Id), Throws.Nothing, $"RetrieveOneAsync should have been called");
+        }
+
+        [Test]
+        public async Task GetToDoAsync_DoesNotExists_NotFoundReturned()
+        {
+            // Arrange
+            const int itemIndex = 0;
+            var expectedToDo = _toDoList.ElementAt(itemIndex);
+            _retrieveServiceSubstitute.IsInDatabaseAsync(expectedToDo.Id).Returns(false);
+
+            // Act
+            var response = await _controller.ExecuteAction(controller => controller.GetToDoAsync(expectedToDo.Id));
+
+            // Assert
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.NotFound), $"Expecting status code NotFound, but was {response.StatusCode}");
+            Assert.That(() => _retrieveServiceSubstitute.Received().RetrieveOneAsync(expectedToDo.Id), Throws.TypeOf<ReceivedCallsException>(), $"RetrieveOneAsync should not have been called");
+        }
+
+        [Test]
+        public async Task PostToDoAsync_CorrectModel_CreatedReturned()
+        {
+            // Arrange
+            const int itemIndex = 2;
+            var expectedToDo = _toDoList.ElementAt(itemIndex);
+            string location = $"todos/{expectedToDo.Id}";
+            var postToDo = Substitute.For<ToDoViewModel>();
```

Comment:
new ToDoViewModel { Text = <what-ever> } seems to be more fitting that substitution here - after all, was the `Convert` more complex, you would test it and could use it anyway. 

Note: Problem with your substitute is that it actually lies a bit since it returns complete `ToDo` item, not only item with the `Text` property filled.